### PR TITLE
Windows: uninstall msi-installed Salt

### PR DESCRIPTION
**Pull request by marbx**
### What does this PR do?
On Windows, uninstall msi-installed Salt

### CODE REVIEW

File: pkg/windows/installer/Salt-Minion-Setup.nsi [hunk 1]
```diff
@@ -902,6 +923,20 @@ Function MsiQueryProductState
 
 FunctionEnd
 
+###############################################################################
+# Source    https://nsis.sourceforge.io/Uninstalling_a_previous_MSI_(Windows_installer_package)
+###############################################################################
+Function UninstallMSI
+    ; $R0 === product code
+    MessageBox MB_YESNOCANCEL|MB_ICONQUESTION  "Uninstalling a previous version of ${PRODUCT_NAME}" IDNO UninstallMSI_nomsi IDYES UninstallMSI_yesmsi
```
[the remark this answers is not in the document]
Comment (twangboy) [reply pkg/windows/installer/Salt-Minion-Setup.nsi:931]: Actually I think the logic here needs to be more like this:
```
Function UninstallMSI
    ; $R0 === product code
    MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION \
        "${PRODUCT_NAME} is already installed.$\n$\n\
        Click `OK` to remove the existing installation." \
        /SD IDOK IDOK UninstallMSI
    Abort

    UninstallMSI:
        ExecWait '"msiexec.exe" /x $R0 /qn /quiet /norestart'

FunctionEnd
```

[the remark this answers is not in the document]
Comment (twangboy) [reply pkg/windows/installer/Salt-Minion-Setup.nsi:931]: I don't think we need to prompt them when the uninstall is completed. It should just continue.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: pkg/windows/installer/Salt-Minion-Setup.nsi (hunk 1 above)
[the remark this answers is not in the document]
Comment (markuskramerIgitt) [reply pkg/windows/installer/Salt-Minion-Setup.nsi]: Committed your logic. 
Please feel free to change the PR yourself, if you see more room for improvement.

[the remark this answers is not in the document]
Comment (markuskramerIgitt) [reply pkg/windows/installer/Salt-Minion-Setup.nsi]: I committed your logic

**Comment by markuskramerIgitt** [comment]
Why the branch has a conflict?

**Comment by markuskramerIgitt** [comment]
Replaced by https://github.com/saltstack/salt/pull/57068 due to merge conflicts

**Pull request closed without merging**